### PR TITLE
Fix build issues with libc mem* benchmarks

### PR DESCRIPTION
**Pull request by dmpots**
Fix a few issues found when trying to build the benchmark:

Errors

1. Unable to find include "src/__support/macros/config.h" in LibcMemoryBenchmarkMain.cpp

Warnings

2. Unused variable warning `Index` in MemorySizeDistributions.cpp
3. Fix deprecation warning for const-ref version of `DoNotOptimize`. warning: 'DoNotOptimize<void *>' is deprecated: The const-ref version of this method can permit undesired compiler optimizations in benchmarks

### CODE REVIEW

File: libc/benchmarks/MemorySizeDistributions.cpp [hunk 1]
```diff
@@ -185,11 +185,9 @@ ArrayRef<MemorySizeDistribution> getMemcmpSizeDistributions() {
 MemorySizeDistribution
 getDistributionOrDie(ArrayRef<MemorySizeDistribution> Distributions,
                      StringRef Name) {
-  size_t Index = 0;
   for (const auto &MSD : Distributions) {
     if (MSD.Name == Name)
       return MSD;
-    ++Index;
   }
```
Comment (nickdesaulniers) [inline libc/benchmarks/MemorySizeDistributions.cpp:191]: Feel free to remove the `{}` on the `for` loop now.
https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements

**Review by nickdesaulniers — APPROVED** [review]

File: libc/benchmarks/MemorySizeDistributions.cpp (hunk 1 above)
Comment (dmpots) [reply libc/benchmarks/MemorySizeDistributions.cpp]: Fixed in latest commit.

**Review by nickdesaulniers — APPROVED** [review]
LGTM; do you need me to merge this for you?

**Comment by dmpots** [comment]
> LGTM; do you need me to merge this for you?

Yes, please! I do not have commit access. Thanks!

**Pull request merged**